src/JBBuybackHook.sol

```diff
@@ -35,72 +37,66 @@ import {IWETH9} from "./interfaces/external/IWETH9.sol";
 /// route.
 /// @dev Compatible with any `JBTerminal` and any project token that can be pooled on Uniswap v3.
 contract JBBuybackHook is JBPermissioned, IJBBuybackHook {
-    //*********************************************************************//
-    // --------------------------- custom errors ------------------------- //
-    //*********************************************************************//
-
-    error SpecifiedSlippageExceeded();
-    error InsufficientPayAmount();
-    error NoProjectToken();
-    error PoolAlreadySet();
-    error InvalidTwapSlippageTolerance();
-    error InvalidTwapWindow();
-    error Unauthorized();
+    // A library that adds default safety checks to ERC20 functionality.
+    using SafeERC20 for IERC20;
 
     //*********************************************************************//
-    // --------------------- internal stored properties ------------------ //
+    // --------------------------- custom errors ------------------------- //
     //*********************************************************************//
 
-    /// @notice The TWAP parameters used for the given project when the payer does not specify a quote.
-    /// See the README for further information.
-    /// @dev This includes the TWAP slippage tolerance and TWAP window, packed into a `uint256`.
-    /// @custom:param projectId The ID of the project to get the twap parameters for.
-    mapping(uint256 projectId => uint256) internal twapParamsOf;
+    error JBBuybackHook_CallerNotPool(address caller);
+    error JBBuybackHook_InsufficientPayAmount(uint256 swapAmount, uint256 totalPaid);
+    error JBBuybackHook_InvalidTwapSlippageTolerance(uint256 value, uint256 min, uint256 max);
+    error JBBuybackHook_InvalidTwapWindow(uint256 value, uint256 min, uint256 max);
+    error JBBuybackHook_PoolAlreadySet(IUniswapV3Pool pool);
+    error JBBuybackHook_SpecifiedSlippageExceeded(uint256 amount, uint256 minimum);
+    error JBBuybackHook_Unauthorized(address caller);
```

Comment:
nit: I feel like the `JBBuybackHook_`  prefix doesn't really make sense for the `Unauthorized` error. Since its used in a lot of places and this would make it so each contract gets its own signature for this error, for UIs it feels more comfortable if we use the same error(s) in lots of places.

Reply:
Makes sense. 
counterpoint would be that a UI would want to know _which_ unauthorized is being triggered, else it’s unclear where the error comes from.